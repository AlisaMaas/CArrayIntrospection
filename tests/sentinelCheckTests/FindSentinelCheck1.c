void print(char* string){}
int find(char string[])
{
  for(int i = 0; ; i++){
		if(string[i] == '\0'){
            print("About to break from loop");
			break;
			}
	}
  return 1;
}
