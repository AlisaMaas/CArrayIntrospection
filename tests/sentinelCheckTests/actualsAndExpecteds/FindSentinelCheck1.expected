Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'print':
Analyzing function: print
	We found: 0 loops
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'foo':
Analyzing function: foo
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.cond
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'find':
Analyzing function: find
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.cond
